### ISSUE TITLE
Reconnect / offline

### ISSUE THREAD

**Issue by zooldk** [node-xmpp/client#199]
FYI: issue moved form https://github.com/node-xmpp/core/issues/89 to here.

Hi Guys,

I've been experimenting a bit with this library and using it as a XMPP client.
I have experienced a lot of problems during disconnection of the networks.

E.g. when I disconnect I expect it to either give me an event "disconnect" or "offline" that I can react on. It does not happen when is do the disconnect.

I have also tried to enable reconnect - setting the reconnect = true on the options, that does not work either.

I have been using BOSH as transport and all my other clients (written in other languages) seems to be working - so it seems like a client implementation problem.

Basically I am trying to "catch" the disconnection with:

```
client.on('offline', function(data) {
    console.log('\n\n' + new Date() + ' offline. \n\n Data: ', data)
})

client.on('error', function(error) {
    console.log('\n\n' + new Date() + ' Error: ', error)
})

client.on('disconnect', function(e) {
    console.log('\n\n' + new Date() + ' Disconnect: ', e)  
})
```

I am using version 3.0.0 of node-xmpp-client.

-Cheers!

/Steffen

**Comment by sonnyp** [comment]
@zooldk define 'disconnect' please

unplugging the cable? calling client.end()? disabling WiFi? ...

**Comment by zooldk** [comment]
Yes. When unplugging. I have updated my node.js and will try again soon. Will get back.

**Comment by zooldk** [comment]
ok. So tried again. And when I have unplugged (no wifi) - then nothing happens. I have waited for a while for a TCP timeout etc. but nothing happens. 

The funny thing is that I am receiving the "disconnect" event when I have reconnected my wifi.

Here is my little test (BOSH connection):
```
$ node test.js 

{ reconnect: true,
  jid: 'xxx.yy/blah',
  password: 'zzzzzzzzz',
  host: 't-xxx.yy',
  bosh: { url: 'http://t-xxx.yy:5280/http-bind' } }


>>Wed Apr 06 2016 12:21:27 GMT+0200 (CEST) online.

 Data:  { jid: 
   JID {
     _local: 'admin',
     user: 'admin',
     _domain: 'xmpp.yousee.tv',
     _resource: 'blah' } }

>>Wed Apr 06 2016 12:21:27 GMT+0200 (CEST)  <<<<<<<<<<<<<  <presence from="xxx.yy/blah" xmlns="jabber:client" to="xxx.yy"/>



// disconnecting for a while... nothing happens...



// re-enabled my wifi and I instantly get the disconnect event.
>>Wed Apr 06 2016 12:23:21 GMT+0200 (CEST)Client is disconnected blah 

```

**Comment by bdipak** [comment]
Same issue with me. Below is my observation:

1.  If I don't set the socket "setKeepAlive" then I don't get any event of internet disconnect/internet connect. My client thinks it is connected as it didn't get any disconnected event.

2. If I set keep alive time then I get disconnected event after internet is connected(ideally it should be on internet disconnection)

PS: I am using lib version: node-xmpp-client: "^2.1.0",

**Comment by zooldk** [comment]
same issue. any news here?
